test/hotspot/jtreg/compiler/intrinsics/math/TestPow0Dot5Opt.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2021 THL A29 Limited, a Tencent company. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8265325
+ * @summary test the optimization of pow(x, 0.5)
+ * @requires os.arch=="amd64" | os.arch=="x86_64"
+ * @run main/othervm TestPow0Dot5Opt
+ * @run main/othervm -Xint TestPow0Dot5Opt
+ * @run main/othervm -Xbatch -XX:TieredStopAtLevel=1 TestPow0Dot5Opt
+ * @run main/othervm -Xbatch -XX:-TieredCompilation  TestPow0Dot5Opt
+ */
+
+public class TestPow0Dot5Opt {
+
+  static void test(double a) throws Exception {
+    // pow(x, 0.5) isn't replaced with sqrt(x) for x < 0.0
+    if (a < 0.0) return;
+
+    double r1 = Math.sqrt(a);
```

Comment:
`r1` should be static value to avoid compiling `sqrt()` method as intrinsic.

Reply:
> `r1` should be static value to avoid compiling `sqrt()` method as intrinsic.

Hi @vnkozlov ,

Thanks for your review.

Patch has been updated based on your comments.

To be honest, I didn't get why r1 should be static value.
I think both static and non-static should be OK for the test.

So what would happen is sqrt() is intrinsified?
Could you please make it more clearer?

Thanks.
Best regards,
Jie

Reply:
Sorry, my bad. I thought to calculate `sqrt(a)` outside `test` method so that only `pow()` code is tested.
Like calculating `gold` value not in compiled code. 
But the test have to use sqrt() for each value `a` and `sqrt` is 1 HW instruction so my suggestion was stupid.
Please, revert this change. Original test code was fine.

Reply:
> Sorry, my bad. I thought to calculate `sqrt(a)` outside `test` method so that only `pow()` code is tested.
> Like calculating `gold` value not in compiled code.
> But the test have to use sqrt() for each value `a` and `sqrt` is 1 HW instruction so my suggestion was stupid.
> Please, revert this change. Original test code was fine.

Thanks for your clarification.
I got your point.

Updated.
Thanks.